File: src/Entity/Voyage.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use Symfony\Component\Serializer\Annotation\Groups;
use Symfony\Component\Serializer\Normalizer\NormalizerIntertace;
/**
 * Voyage
 *
 * @ORM\Table(name="voyage", indexes={@ORM\Index(name="idC", columns={"idC"})})
 * @ORM\Entity
 */

class Voyage
{
    /**
     * @var int
     *
     * @ORM\Column(name="idVoy", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    #[Groups("voyages")]
    private $idvoy;

    /**
     * @ORM\Column(name="dateDVoy", type="string", length=255, nullable=false)
     * @Assert\Date
     */
    #[Groups("voyages")]
    private $datedvoy;

    /**
     * @ORM\Column(name="dateRVoy", type="string", length=255, nullable=false)
     * @Assert\Date
     * @Assert\Expression(
     *     "this.getDatedvoy() < this.getDatervoy()",
     *     message="La date de fin doit être après la date de début"
     * )
     */
    #[Groups("voyages")]
    private $datervoy;

    /**
     * @ORM\Column(name="destination", type="string", length=255, nullable=false)
     * @Assert\Length(min=5)
     * @Assert\Regex(
     *     pattern="/^[a-zA-Z]+$/",
     *     message="Le destination ne doit contenir que des lettres"
     * )
     */
    #[Groups("voyages")]
    private $destination;

    /**
     * @var \Contrat
     *
     * @ORM\ManyToOne(targetEntity="Contrat")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="idC", referencedColumnName="idC")
     * })
     */
    #[Groups("voyages")]
    private $idc;

    public function getIdvoy(): ?int
    {
        return $this->idvoy;
    }

    public function getDatedvoy(): ?string
    {
        return $this->datedvoy;
    }

    public function setDatedvoy(string $datedvoy): self
    {
        $this->datedvoy = $datedvoy;

        return $this;
    }

    public function getDatervoy(): ?string
    {
        return $this->datervoy;
    }

    public function setDatervoy(string $datervoy): self
    {
        $this->datervoy = $datervoy;

        return $this;
    }

    public function getDestination(): ?string
    {
        return $this->destination;
    }

    public function setDestination(?string $destination): self
    {
        $this->destination = $destination;

        return $this;
    }

    public function getIdc(): ?Contrat
    {
        return $this->idc;
    }

    public function setIdc(?Contrat $idc): self
    {
        $this->idc = $idc;

        return $this;
    }
    public function __toString(): string
    {
        return $this->destination;
    }


}

File: AdoDocMobile/src/com/mycompany/myapp/entities/Voyage.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mycompany.myapp.entities;
import java.util.Date;
/**
 *
 * @author ASMA
 */
public class Voyage {
    private int idVoy;
    private String dateDvoy,dateRvoy,destination;
    private int idC;

    public Voyage(String dateDvoy, String dateRvoy, String destination, int idC) {
        this.dateDvoy = dateDvoy;
        this.dateRvoy = dateRvoy;
        this.destination = destination;
        this.idC = idC;
    }

    
    
    public Voyage(String destination) {
        this.destination = destination;
    }

    public Voyage() {
    }

    public Voyage(int idVoy, String dateDvoy, String dateRvoy, String destination, int idC) {
        this.idVoy = idVoy;
        this.dateDvoy = dateDvoy;
        this.dateRvoy = dateRvoy;
        this.destination = destination;
        this.idC = idC;
    }

    public Voyage(String dateDvoy, String dateRvoy, String destination) {
        this.dateDvoy = dateDvoy;
        this.dateRvoy = dateRvoy;
        this.destination = destination;
    }

    public int getIdVoy() {
        return idVoy;
    }

    public String getDateDvoy() {
        return dateDvoy;
    }

    public String getDateRvoy() {
        return dateRvoy;
    }

    public String getDestination() {
        return destination;
    }

    public int getIdC() {
        return idC;
    }

    public void setIdVoy(int idVoy) {
        this.idVoy = idVoy;
    }

    public void setDateDvoy(String dateDvoy) {
        this.dateDvoy = dateDvoy;
    }

    public void setDateRvoy(String dateRvoy) {
        this.dateRvoy = dateRvoy;
    }

    public void setDestination(String destination) {
        this.destination = destination;
    }

    public void setIdC(int idC) {
        this.idC = idC;
    }

    @Override
    public String toString() {
        return "Voyage{" + "idVoy=" + idVoy + ", dateDvoy=" + dateDvoy + ", dateRvoy=" + dateRvoy + ", destination=" + destination + ", idC=" + idC + '}';
    }
    
    
}
